src/main/java/com/minecolonies/coremod/util/BlockUtils.java

```diff
@@ -185,68 +186,9 @@ public static boolean isWater(final IBlockState iBlockState)
      */
     public static boolean isBlockSeed(@NotNull final World world, @NotNull final BlockPos pos)
     {
-        final ItemStack stack = BlockUtils.getItemStackFromBlockState(world.getBlockState(pos.up()));
-        return  stack != null && stack.getItem() instanceof ItemSeeds;
-    }
-
-    /**
-     * Mimics pick block.
-     *
-     * @param blockState the block and state we are creating an ItemStack for.
-     * @return ItemStack fromt the BlockState.
-     */
-    public static ItemStack getItemStackFromBlockState(@NotNull final IBlockState blockState)
-    {
-        final Item item = getItem(blockState);
-        if (item == null)
-        {
-            if(blockState.getBlock() instanceof BlockFire)
-            {
-                return null;
-            }
-
-            return null;
-        }
-
-        Block block = blockState.getBlock();
-        if (item instanceof ItemBlock)
-        {
-            block = Block.getBlockFromItem(item);
-        }
-
-        return new ItemStack(item, 1, getDamageValue(block, blockState));
+        return BlockUtils.getItem(world.getBlockState(pos.up())) instanceof ItemSeeds;
     }
 
-    /**
-     * Compares two blocks and checks if they are equally dirt.
-     * Meaning dirt and grass are equal. But podzol and coarse dirt not.
-     *
-     * @param structureBlock the block of the structure.
-     * @param worldBlock the world block.
-     * @param structureMetaData the structure metadata.
-     * @param worldMetadata the world metadata.
-     * @return true if equal.
-     */
-    public static boolean isGrassOrDirt(@NotNull final Block structureBlock, @NotNull final Block worldBlock,
-            @NotNull final IBlockState structureMetaData, @NotNull final IBlockState worldMetadata)
-    {
-        if((structureBlock == Blocks.DIRT || structureBlock == Blocks.GRASS) && (worldBlock == Blocks.DIRT || worldBlock == Blocks.GRASS))
-        {
-            if(structureBlock == Blocks.DIRT
-                    && (structureMetaData.getValue(BlockDirt.VARIANT) == BlockDirt.DirtType.COARSE_DIRT
-                    || structureMetaData.getValue(BlockDirt.VARIANT) == BlockDirt.DirtType.PODZOL))
-            {
-                return false;
-            }
-
-            return worldBlock != Blocks.DIRT
-                    || (worldMetadata.getValue(BlockDirt.VARIANT) != BlockDirt.DirtType.COARSE_DIRT
-                    && worldMetadata.getValue(BlockDirt.VARIANT) != BlockDirt.DirtType.PODZOL);
-        }
-        return  false;
-    }
-
-
     private static Item getItem(@NotNull final IBlockState blockState)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This method has 150 lines, which is greater than the 100 lines authorized. Split it into smaller methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS138)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 34, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "getItem" is 43 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 68 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)